templates/walle.py

```diff
@@ -422,40 +444,113 @@ def get_running_jobs(self, tool=None) -> [Job]:
 
 
+def get_path_from_env_or_error(env: str) -> pathlib.Path:
+    if os.environ.get(env):
+        if (path := pathlib.Path(os.environ.get(env).strip())).exists():
+            return path
+        else:
+            raise ValueError(f"Path for {env} is invalid")
+    else:
+        raise ValueError(f"Please set ENV {env}")
+
+
+def get_str_from_env_or_error(env: str) -> str:
+    if os.environ.get(env):
+        if len(from_env := os.environ.get(env).strip()) > 0:
+            return from_env
+        else:
+            raise ValueError(f"Path for {env} is invalid")
+    else:
+        raise ValueError(f"Please set ENV {env}")
+
+
+class GalaxyAPI:
+    def __init__(self, base_url: str, api_key: str) -> None:
+        self.base_url = base_url
+        self.api_key = api_key
+        self.auth_header = {"x-api-key": self.api_key}
+
+    def delete_user(self, encoded_user_id: str) -> bool:
+        url = f"{self.base_url}/api/users/{encoded_user_id}"
+        response = requests.delete(url=url, headers=self.auth_header)
+        if response.status_code == 200:
+            return True
+        else:
+            print(f"Failed to delete user {encoded_user_id}!")
+
+    def encode_galaxy_user_id(self, decoded_id: int) -> str:
+        url = f"{self.base_url}/api/configuration/encode/{decoded_id}"
+        response = requests.get(url=url, headers=self.auth_header)
+        if response.status_code != 200:
+            print(f"Failed to encode id {decoded_id}!")
+        else:
+            json_response = response.json()
+            return json_response["encoded_id"]
+
+
 def main():
     """
     Miner Finder's main function. Shows a status bar while processing the jobs found in Galaxy
     """
     args = make_parser().parse_args()
-    jwd_getter = JWDGetter()
-    db = RunningJobDatabase()
-    malware_library = construct_malware_list(load_malware_lib_from_env())
+    galaxy_config_file = get_path_from_env_or_error("GALAXY_CONFIG_FILE")
+
+    jwd_getter = JWDGetter(
+        galaxy_config_file=galaxy_config_file,
+        pulsar_app_conf=get_path_from_env_or_error("GALAXY_PULSAR_APP_CONF"),
+    )
+    db = RunningJobDatabase(
+        db_host=get_str_from_env_or_error("PGHOST"),
+        db_password=galaxy_jwd.extract_password_from_pgpass(
+            get_path_from_env_or_error("PGPASSFILE")
+        ),
+        db_name=get_str_from_env_or_error("PGDATABASE"),
+        db_user=get_str_from_env_or_error("PGUSER"),
+    )
+    malware_library = construct_malware_list(
+        malware_yaml=load_malware_lib_from_env(
+            malware_file=get_path_from_env_or_error("MALWARE_LIB")
+        )
+    )
     jobs = db.get_running_jobs(args.tool)
+    if args.delete_user:
+        api = GalaxyAPI(
+            api_key=get_str_from_env_or_error("GALAXY_API_KEY"),
+            base_url=get_str_from_env_or_error("GALAXY_BASE_URL"),
+        )
+        delete_users = set()
+    if not args.verbose:
+        report_users = set()
```

Comment:
I would not put the variables in the if condition, just fill the set and only filter print() with arg.verbose

Reply:
like this?